src/librustc/traits/error_reporting.rs

```diff
@@ -550,10 +550,33 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                             // error message, let's display it!
                             err.note(&s);
                         } else {
-                            // If we can't show anything useful, try to find
-                            // similar impls.
-                            let impl_candidates = self.find_similar_impl_candidates(trait_ref);
-                            self.report_similar_impl_candidates(impl_candidates, &mut err);
+                            let binops: FxHashMap<&str, &str> = [
```

Comment:
Use lang items please - `tcx.lang_items.add_trait` etc. - instead of guessing stuff.